CONTRIBUTING_A_NEW_QUEST.md

```diff
@@ -0,0 +1,521 @@
+
+- [mdpi](app/src/main/res/drawable-mdpi) - 384 pixels for images, with three square images in each row it would be 128 x 128 pixels for each)
+- [hdpi](app/src/main/res/drawable-hdpi) - 576 pixels for images (192 x 192 pixels in case of three square images in each row)
+- [xhdpi](app/src/main/res/drawable-xhdpi) 768 pixels for images (256 x 256 pixels in case of three square images in each row)
+- [xxhdpi](app/src/main/res/drawable-xxhdpi) 1152 pixels (384 x 384 pixels in case of three square images in each row)
+
+Each of these folders should hold the same image resized to a different resolution. While testing various images it is enough to put one into any of the folders.
+
+The [rescaling script](https://github.com/matkoniecz/rescaling_for_android) may be useful, but you can also do this manually with Gimp or similar software.
+
+After adding a photo, remember to update [the credit file](app/src/main/res/authors.txt) (different to the one for icons).
+
+## Resurvey
+
+Some quests are asked not only when tag is missing but also when it is likely to be outdated. To achieve this `elementFilter` needs to query not only elements missing some tags.
+
+Typical code is in [quest asking about motorcycle parking capacity](app/src/main/java/de/westnordost/streetcomplete/quests/motorcycle_parking_capacity/AddMotorcycleParkingCapacity.kt):
+
+```kotlin
+override val elementFilter = """
+    nodes, ways with amenity = motorcycle_parking
+      and access !~ private|no
+      and (!capacity or capacity older today -4 years)
+"""
+```
+
+This quest will be triggered when:
+
+- `nodes, ways with`
+  - on nodes or ways (relations not eligible)
+- `amenity = motorcycle_parking`
+  - where `amenity = motorcycle_parking` tag is present
+- `and access !~ private|no`
+  - and `access` tag does not have value `private` nor `no`
+- `and (!capacity or capacity older today -4 years)`
+  - and one of following is fullfilled:
+    - `capacity` tag is not present at all (`!capacity`)
+    - element was not edited for a long time (base time is 4 years, but it can be influenced by user changing settings)
+    - `check_date:capacity` with a date indicating that it is outdated (the same as above applies)
+
+## Custom filters
+
+It is possible to use far more complex filters when querying for eligible elements.
+
+Matches like `surface ~ earth|dirt|ground` are possible and are evaluated as "`surface` is either of `earth`, `dirt`, `ground`"
+
+`access !~ private|no` will be evaluated to "`access` is neither `private` nor `no`"
+
+Simple lists are [evaluated as sets](https://github.com/streetcomplete/StreetComplete/blob/2e812b9a3b5288983309a7edde6e8f9db05ad3f2/app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/ElementFilterOverpassKtTest.kt#L79-L88) - and for example `surface ~ earth|dirt|ground` will not match `surface=earther` tag. Basically, `surface ~ earth|dirt|ground` is treated as `surface ~ ^earth|dirt|ground$`
```

Comment:
```suggestion
```

Delete this because as noted before, the information is not relevant and also misleading